### PR TITLE
[sec_scan][17] add `AssertDevice` to `FakeDeviceService`

### PR DESCRIPTION
**Pull request by tigrato**
This PR introduces a `AssertDevice` logic into `FakeDeviceService` to authenticate devices during unit tests using device trust credentials.

This PR is part of https://github.com/gravitational/access-graph/issues/637.

### CODE REVIEW

File: lib/devicetrust/testenv/testenv.go [hunk 1]
```diff
@@ -45,6 +46,25 @@ func WithAutoCreateDevice(b bool) Opt {
...
+		defer e.Service.mu.Unlock()
+		e.Service.devices = append(e.Service.devices,
+			storedDevice{
+				pb:          dev,
+				enrollToken: FakeEnrollmentToken,
+				pub:         pub,
+			},
+		)
+	}
+
```
Comment (codingllama) [inline lib/devicetrust/testenv/testenv.go:65]: nit:

```suggestion
```

File: lib/devicetrust/testenv/fake_device_service.go [hunk 2]
```diff
@@ -483,6 +484,58 @@ func enrollMacOS(stream devicepb.DeviceTrustService_EnrollDeviceServer, initReq
...
+// CreateAssertCeremony creates a fake, server-side device assertion ceremony.
+func (s *FakeDeviceService) CreateAssertCeremony() (assertserver.Ceremony, error) {
+	return s, nil
+}
+
+// AssertDevice implements a fake, server-side device assertion ceremony.
+//
+// AssertDevice requires an enrolled device, so the challenge signature
+// can be verified.
+func (s *FakeDeviceService) AssertDevice(ctx context.Context, stream assertserver.AssertDeviceServerStream) (*devicepb.Device, error) {
```
Comment (codingllama) [inline lib/devicetrust/testenv/fake_device_service.go:496]: This is great, thanks.

We could probably simplify [assert_test.go][1] using this.

[1]: https://github.com/gravitational/teleport/blob/dbe1295cdcf24325ba37be505c51e0b4a8248fa9/lib/devicetrust/assert/assert_test.go#L96

**Review by codingllama — APPROVED** [review]

**Review by jakule — APPROVED** [review]

File: lib/devicetrust/assert/assert_test.go [hunk 3]
```diff
@@ -87,78 +85,83 @@ func TestCeremony(t *testing.T) {
...
+				err := assertC.Run(ctx, &assertStreamClientAdapter{
+					ctx:            ctx,
+					clientToServer: clientToServer,
+					serverToClient: serverToClient,
+				})
+				return trace.Wrap(err, "server AssertDevice errored")
+			})
+
+			serverAssertC, err := env.Service.CreateAssertCeremony()
+			require.NoError(t, err, "CreateAssertCeremony errored")
```
Comment (codingllama) [inline lib/devicetrust/assert/assert_test.go:104]: Do this before the "go func() {}" statements, so we know nothing aborts the tests after the goroutines spawn?

File: lib/devicetrust/assert/assert_test.go [hunk 4]
```diff
@@ -87,78 +85,83 @@ func TestCeremony(t *testing.T) {
...
-			// Typically this would be some other, non-DeviceTrustService stream, but
-			// here this is a good way to test it (as it runs actual fake device
-			// authn)
-			if err := assertC.Run(ctx, &assertStreamAdapter{
-				stream: authnStream,
-			}); err != nil {
-				t.Errorf("Run returned err=%q, want nil", err)
-			}
+			clientToServer := make(chan *devicepb.AssertDeviceRequest, 1)
+			serverToClient := make(chan *devicepb.AssertDeviceResponse, 1)
```
Comment (codingllama) [inline lib/devicetrust/assert/assert_test.go:89]: Suggestion: drop the buffers? We don't really seem to need them.

```suggestion
			clientToServer := make(chan *devicepb.AssertDeviceRequest)
			serverToClient := make(chan *devicepb.AssertDeviceResponse)
```

**Review by codingllama — COMMENTED** [review]
Thanks for the test changes!

**Review by codingllama — APPROVED** [review]

**Pull request merged**